Libraries/LibC/spawn.cpp

```diff
@@ -0,0 +1,250 @@
+    const posix_spawn_file_actions_t* file_actions,
+    const posix_spawnattr_t* RESTRICT attr,
+    char* const argv[RESTRICT], char* const envp[RESTRICT])
+{
+    pid_t child_pid = fork();
+    if (child_pid < 0)
+        return errno;
+
+    if (child_pid != 0) {
+        *out_pid = child_pid;
+        return 0;
+    }
+
+    posix_spawn_child(path, file_actions, attr, argv, envp, /*search_path=*/true);
+}
+
+
+#if 0
+// FIXME
+int posix_spawn_file_actions_addclose(posix_spawn_file_actions_t*,
+    int)
+{
+}
+
+int posix_spawn_file_actions_adddup2(posix_spawn_file_actions_t*,
+    int, int)
+{
+}
+
+int posix_spawn_file_actions_addopen(posix_spawn_file_actions_t* RESTRICT,
+    int, const char* RESTRICT, int, mode_t)
+{
+}
+
+int posix_spawn_file_actions_destroy(posix_spawn_file_actions_t*)
+{
+}
+
+int posix_spawn_file_actions_init(posix_spawn_file_actions_t*)
+{
+}
+#endif
+
+
+int posix_spawnattr_destroy(posix_spawnattr_t*)
+{
+    return 0;
+}
+
+int posix_spawnattr_getflags(const posix_spawnattr_t* RESTRICT attr,
+    short* RESTRICT out_flags)
+{
+    *out_flags = attr->flags;
+    return 0;
+}
+
+int posix_spawnattr_getpgroup(const posix_spawnattr_t* RESTRICT attr,
+    pid_t* RESTRICT out_pgroup)
+{
+    *out_pgroup = attr->pgroup;
+    return 0;
+}
+
+int posix_spawnattr_getschedparam(const posix_spawnattr_t* RESTRICT attr,
+    struct sched_param* RESTRICT out_schedparam)
+{
+    *out_schedparam = attr->schedparam;
+    return 0;
+}
+
+int posix_spawnattr_getschedpolicy(const posix_spawnattr_t* RESTRICT attr,
+    int* RESTRICT out_schedpolicty)
+{
+    *out_schedpolicty = attr->schedpolicy;
+    return 0;
+}
+
+int posix_spawnattr_getsigdefault(const posix_spawnattr_t* RESTRICT attr,
+    sigset_t* RESTRICT out_sigdefault)
+{
+    *out_sigdefault = attr->sigdefault;
+    return 0;
+}
+
+int posix_spawnattr_getsigmask(const posix_spawnattr_t* RESTRICT attr,
+    sigset_t* RESTRICT out_sigmask)
+{
+    *out_sigmask = attr->sigmask;
+    return 0;
+}
+
+int posix_spawnattr_init(posix_spawnattr_t* attr)
+{
+    attr->flags = 0;
+    attr->pgroup = 0;
+    // attr->schedparam intentionally not written; its default value is unspecified.
+    // attr->schedpolicy intentionally not written; its default value is unspecified.
+    sigemptyset(&attr->sigdefault);
+    // attr->sigmask intentionally not written; its default value is unspecified.
+    return 0;
+}
+
+int posix_spawnattr_setflags(posix_spawnattr_t* attr, short flags)
+{
+    if (flags & ~(POSIX_SPAWN_RESETIDS | POSIX_SPAWN_SETPGROUP | POSIX_SPAWN_SETSCHEDPARAM | POSIX_SPAWN_SETSCHEDULER | POSIX_SPAWN_SETSIGDEF | POSIX_SPAWN_SETSIGMASK))
+        return EINVAL;
+
+    attr->flags = flags;
+    return 0;
+}
+
+int posix_spawnattr_setpgroup(posix_spawnattr_t* attr, pid_t pgroup)
+{
+    attr->pgroup = pgroup;
+    return 0;
+}
+
+int posix_spawnattr_setschedparam(posix_spawnattr_t* RESTRICT attr,
+    const struct sched_param* RESTRICT schedparam)
```

Comment:
Pls turn off "wrap all lines at 80 columns" or whatever causes this :)

Reply:
I copied it over from  https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/spawn.h.html and ran clang-format. Am I supposed to join the whole file into a single line before running clang-format?

(Will do a best guess at what you want, but not 100% sure.)